L1Trigger/Phase2L1ParticleFlow/src/egamma/L1EGPuppiIsoAlgo.cc

```diff
@@ -0,0 +1,138 @@
+#include "L1Trigger/Phase2L1ParticleFlow/interface/egamma/L1EGPuppiIsoAlgo.h"
+
+using namespace l1ct;
+
+L1EGPuppiIsoAlgo::L1EGPuppiIsoAlgo(const edm::ParameterSet& pSet)
+    : config_(pSet.getParameter<std::string>("pfIsoType"),
+              pSet.getParameter<double>("pfPtMin"),
+              pSet.getParameter<double>("dZ"),
+              pSet.getParameter<double>("dRMin"),
+              pSet.getParameter<double>("dRMax"),
+              pSet.getParameter<bool>("pfCandReuse")) {}
+
+void L1EGPuppiIsoAlgo::run(const EGIsoObjsEmu& l1EGs,
+                           const PuppiObjs& l1PFCands,
+                           EGIsoObjsEmu& outL1EGs,
+                           z0_t z0) const {
+  outL1EGs.reserve(l1EGs.size());
+
+  // make a list of pointers to PF candidates
+  // the pointer will be removed from the list once the candidate has been used
+  std::list<std::shared_ptr<PuppiObj>> workPFCands;
+  std::list<std::shared_ptr<PuppiObj>> workPFCandsPV;
```

Comment:
why do you need `shared_ptr`?
if you never modify the puppi candidates themselves, only remove them from the list, then you could have `const PuppiObj *`  instead.
(using a `std::vector` and setting the erased candidates to `nullptr` may also be more efficient)

Reply:
Mainly not to use bare pointers. But in this case it is actually OK as you say. Changed.

Concerning using a `std::vector` instead I guess it depends on the overlap of the cones. The `std::list` will get shorter fro subsequent EG objects while the `sd::vector` stays the same and one has to additionally check for the `nullptr`. In any case I do not think, after initial tests with few events, that it has a huge impact.

Reply:
@gpetruc FYI